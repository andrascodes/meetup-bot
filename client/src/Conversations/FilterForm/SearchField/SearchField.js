import React from 'react'

export const SearchField = props => {
  
  const listOptions = props.options.map(person => <option key={person} value={person} />)

  return (
    <div className="form-group SearchField">
      <label htmlFor="participants" className="SearchLabel">Participant:</label>
      <input list="participants" className="SearchInput" 
        onInput={props.handleInput} 
        defaultValue={props.defaultValue}
      />
      <datalist id="participants">
        {listOptions}
      </datalist>
    </div>
  )
}

SearchField.propTypes = {
  options: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
  handleInput: React.PropTypes.func.isRequired,
  defaultValue: React.PropTypes.string
}