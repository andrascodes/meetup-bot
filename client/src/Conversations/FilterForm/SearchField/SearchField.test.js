import React from 'react'
import renderer from 'react-test-renderer'

import { SearchField } from './SearchField'

describe('SearchField', () => {

  test('renders correctly', () => {
    
    const tree = renderer.create(
      <SearchField 
        options={['123', '456']}
        handleInput={() => {}}
        defaultParticipantValue={'123'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('renders correctly when no defaultValue is set', () => {
    
    const tree = renderer.create(
      <SearchField 
        options={['123', '456']}
        handleInput={() => {}}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('throws PropType error', () => {
    const render = () => renderer.create(
      <SearchField
        handleInput={() => {}}
      />
    )
    expect(render).toThrowError(`The prop 'options' has to be defined, pass an empty array ([])`)
  })

})