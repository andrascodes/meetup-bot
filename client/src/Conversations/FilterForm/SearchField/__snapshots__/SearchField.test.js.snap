exports[`SearchField renders correctly 1`] = `
<div
  className="form-group SearchField">
  <label
    className="SearchLabel"
    htmlFor="participants">
    Participant:
  </label>
  <input
    className="SearchInput"
    defaultValue="123"
    list="participants"
    onInput={[Function]} />
  <datalist
    id="participants">
    <option
      value="123" />
    <option
      value="456" />
  </datalist>
</div>
`;

exports[`SearchField renders correctly when no defaultValue is set 1`] = `
<div
  className="form-group SearchField">
  <label
    className="SearchLabel"
    htmlFor="participants">
    Participant:
  </label>
  <input
    className="SearchInput"
    defaultValue={undefined}
    list="participants"
    onInput={[Function]} />
  <datalist
    id="participants">
    <option
      value="123" />
    <option
      value="456" />
  </datalist>
</div>
`;
